### PR TITLE
switch from dep to go modules

### PR DESCRIPTION
**Pull request by kubasobon**
## Checklist
- [x] Wait for required modules to be migrated to go modules:
    - [x] apprclient
    - [x] errors
    - [x] helmclient
- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
+
+workflows:
+  test:
+    jobs:
+      - architect/go-test:
+          name: go-test
+          filters:
+            # Trigger job also on git tag.
+            tags:
+              only: /^v.*/
```
Comment (tomahawk28) [inline .circleci/config.yml:14]: I don't mind but usually, we do it by another PR, don't we?

Comment (kubasobon) [reply .circleci/config.yml:14]: I know. Got excited after figuring out the `go mod tidy` issue.

**Review by tomahawk28 — APPROVED** [review]
